File: 2015/02-20/Pat/Challenge1.java
/**
 * @author:     Pat Hawks
 * Created on:  Feb 02, 2015
 * Source File: Challenge1.java
 *
 * To compile:
 *    javac -Xlint:all Challenge1.java
 */

import java.util.Scanner;

public class Challenge1 {
    public static void main( String[] args ) {
        final String PROMPT_STRING = "Enter a whole number between %d and %d "
                                   + "without spaces or commas.\n";
        final String FORMAT_STRING = "%,d\n";
        final int MIN_NUM = 1000, MAX_NUM = 999999;
        Scanner in = new Scanner( System.in );
        int inNum;

        do {
            System.out.printf( PROMPT_STRING, MIN_NUM, MAX_NUM );
            inNum = in.nextInt();
        } while ( inNum < MIN_NUM || inNum > MAX_NUM );

//      The Right Way
        System.out.printf( FORMAT_STRING, inNum );

//      System.out.printf( "%d,%3d\n", inNum/1000, inNum%1000 );

/*      System.out.print( inNum/1000 );
 *      System.out.print( "," );
 *      if( inNum%1000 < 100 ) // Print some leading zeros?
 *          System.out.print( inNum%1000<10 ? "00" : "0" );
 *      System.out.println( inNum%1000 );
 */
        in.close();
        return;
    }
}

File: 2015/02-20/Pat/Challenge1.cpp
/**
 * @author:     Pat Hawks
 * Created on:  Feb 02, 2015
 * Source File: Challenge1.cpp
 *
 * To compile:
 *    g++ Challenge1.cpp -Wall -o Challenge1
 */

#include <iomanip>
#include <iostream>
#include <locale>
#include <string>
using namespace std;

struct formatNumber : numpunct<char> {
    string do_grouping() const {return "\03";}
};

int main( void ) {
    const string PROMPT_STRING =
        "Enter a whole number between 1000 and 999999 without spaces or commas.";
    int inNum;
    do {
        cout << PROMPT_STRING << endl;
        cin >> inNum;
    } while ( inNum < 1000 || inNum > 999999 );

    locale loc( cout.getloc(), new formatNumber );
    cout.imbue( loc );
    cout << inNum << endl;

    return 0;
}
